README.md

```diff
@@ -0,0 +1,129 @@
+# StackStorm in Docker containers
+
+The initial container configuration is as follows:
+
+ - st2-upstart (st2 + st2web + sshd)
+ - mongo
+ - rabbitmq
+
+## Usage
+
+Build the st2-upstart image:
+
+  ```
+  make build
+  ```
+
+Start the docker environment:
+
+  ```
+  docker-compose up
+  ```
+
+Install the SSH keys locally, and setup ST2 user (this only needs to be run once per `docker-compose up`):
+
+*NOTE: The default values for `ST2USER` and `ST2PASSWORD` are `st2admin` and `Ch@ngeMe`
+respectively. You only need to specify these variables if you do not want to use the default
+values.*
+
+  ```
+  make setup ST2USER=st2admin ST2PASSWORD=Ch@ngeMe
+  ```
+
+Use either `ssh` or `docker exec` to connect to the st2-upstart container:
+
+  ```
+  ssh -i ~/.ssh/id_busybee root@localhost
+  ```
+
+  ```
+  docker exec -it st2docker_st2-upstart_1 /bin/bash
+  ```
+
+Once connected to the st2-upstart container, verify that `st2-self-check` passes successfully.
+
+*NOTE: For now, the mistral tests will fail because st2mistral is not yet configured.*
+
+  ```
+  st2ctl reload
+  . ~/st2.vars
+  st2-self-check
+  ```
+
+*NOTE: The `./packs` directory is mounted into the `st2-upstart` container at `/opt/stackstorm/packs`.*
+
+To overwrite `./packs` with the packs provided by st2, run:
+
+  ```
+  rm -rf /opt/stackstorm/packs/*
+  cp -R /opt/stackstorm/packs.pkg/* /opt/stackstorm/packs
+  ```
+
+To stop the docker environment, run:
+
+  ```
+  docker-compose down
+  ```
+
+## Adding a new action
+
+As an example of how to create a new action, let's add a new action called `echo_action`.
+
+First, on the host, we create the metadata file `./packs/packs/actions/my_echo_action.yaml`:
```

Comment:
I am +1 to using either python or shell script as the first action example. They're both going to require both the script and the metadata file.